pkg/test-infra/tiflash/template.go

```diff
@@ -0,0 +1,194 @@
+          sed s/{pod_num}/${ordinal}/g /etc/tiflash/proxy_templ.toml > /data/proxy.toml
+	`
+)
+
+type tiFlashConfig struct {
+	ClusterName string
+	Namespace   string
+}
+
+func renderTiFlashConfig(model *tiFlashConfig) (string, error) {
+	return renderTemplateFunc(tiFlashTpl, model)
+}
+
+var tiFlashTpl = template.Must(template.New("tiflash-config").Parse(`tmp_path = "/data/tmp"
+    display_name = "TiFlash"
+    default_profile = "default"
+    users_config = "/etc/tiflash/users.toml"
+    path = "/data/db"
+    mark_cache_size = 5368709120
+    listen_host = "0.0.0.0"
+    tcp_port = 9000
+    http_port = 8123
+    interserver_http_port = 9009
+
+    [flash]
+    tidb_status_addr = "{{.ClusterName}}-tidb.{{.Namespace}}.svc.cluster.local:10080"
+	service_addr = "{{.ClusterName}}-tiflash-{pod_num}.{{.ClusterName}}-tiflash.{{.Namespace}}.svc.cluster.local:3930"
+    overlap_threshold = 0.6
+    compact_log_min_period = 10
+
+    [flash.flash_cluster]
+    master_ttl = 60
+    refresh_interval = 20
+    update_rule_interval = 5
+    cluster_manager_path = "/flash_cluster_manager"
+
+    [flash.proxy]
+    addr = "0.0.0.0:20170"
+    data-dir = "/data/proxy"
+    config = "/data/proxy.toml"
+    log-file = "/data/logs/proxy.log"
+
+    [logger]
+    level = "debug"
+    log = "/data/logs/server.log"
+    errorlog = "/data/logs/error.log"
+    size = "4000M"
+    count = 10
+
+    [application]
+    runAsDaemon = true
+
+    [raft]
+    kvstore_path = "/data/kvstore"
+    pd_addr = "{{.ClusterName}}-pd.{{.Namespace}}.svc.cluster.local:2379"
+    storage_engine = "dt"
+
+    [status]
+    metrics_port = 8234
+`))
+
+func renderTiFlashProxyTpl(model *tiFlashConfig) (string, error) {
+	return renderTemplateFunc(tiFlashProxyTpl, model)
+}
+
+var tiFlashProxyTpl = template.Must(template.New("tiflash-config").Parse(`log-level = "info"
+
+    [readpool.storage]
+
+    [readpool.coprocessor]
+
+    [server]
+    engine-addr = "{{.ClusterName}}-tiflash-{pod_num}.{{.ClusterName}}-tiflash.{{.Namespace}}.svc.cluster.local:3930"
+    advertise-addr = "{{.ClusterName}}-tiflash-{pod_num}.{{.ClusterName}}-tiflash.{{.Namespace}}.svc.cluster.local:20170"
+
+    [storage]
+
+    [pd]
+
+    [metric]
+
+    [raftstore]
+    raftdb-path = ""
+    sync-log = true
+    #max-leader-missing-duration = "22s"
+    #abnormal-leader-missing-duration = "21s"
+    #peer-stale-state-check-interval = "20s"
+    hibernate-regions = false
+
+    [coprocessor]
+
+    [rocksdb]
+    wal-dir = ""
+    max-open-files = 1000
+
+    [rocksdb.defaultcf]
+    block-cache-size = "10GB"
+
+    [rocksdb.lockcf]
+    block-cache-size = "4GB"
+
+    [rocksdb.writecf]
+    block-cache-size = "4GB"
+
+    [raftdb]
+    max-open-files = 1000
+
+    [raftdb.defaultcf]
+    block-cache-size = "1GB"
+
+    [security]
+    ca-path = ""
+    cert-path = ""
+    key-path = ""
+
+    [import]
+
+`))
+
+func renderTemplateFunc(tpl *template.Template, model interface{}) (string, error) {
```

Comment:
How about extract it into a public method? I see many of renderTemplateFunc on our codebase.